src/uint256.h

```diff
@@ -105,6 +126,7 @@ class uint160 : public base_blob<160> {
  */
 class uint256 : public base_blob<256> {
 public:
+    static std::optional<uint256> FromHex(std::string_view str) { return detail::FromHex<uint256>(str); }
```

Comment:
Why not have all `FromHex` methods be `[[nodiscard]]`? Seems bad to throw away work.

Reply:
In principle all pure functions like `FromHex` (that at most mutate input-parameters and don't have asserts/throw) should be `[[nodiscard]]`. One could at least apply that rule to new code?